packages/components/src/components/data-list/data-list.tsx

```diff
@@ -0,0 +1,244 @@
+import classNames from 'classnames';
+import React from 'react';
+import { TransitionGroup } from 'react-transition-group';
+import {
+    List as _List,
+    CellMeasurer as _CellMeasurer,
+    CellMeasurerCache,
+    CellMeasurerProps,
+    AutoSizer as _AutoSizer,
+    type AutoSizerProps,
+    ScrollParams,
+    ListProps,
+    ListRowProps,
+} from 'react-virtualized';
+import { isMobile, isDesktop } from '@deriv/shared';
+import DataListCell from './data-list-cell';
+import DataListRow from './data-list-row';
+import ThemedScrollbars from '../themed-scrollbars';
+import { MeasuredCellParent } from 'react-virtualized/dist/es/CellMeasurer';
+
+const List = _List as unknown as React.FC<ListProps>;
+const AutoSizer = _AutoSizer as unknown as React.FC<AutoSizerProps>;
+const CellMeasurer = _CellMeasurer as unknown as React.FC<CellMeasurerProps>;
+type CellType = { Cell: typeof DataListCell };
+export type TRowRenderer = (params: { row: any; is_footer?: boolean; measure?: () => void }) => React.ReactNode;
+type DataListProps = {
+    className?: string;
+    data_source: any;
+    footer?: React.ReactNode;
+    getRowAction?: (row: React.ReactNode) => string;
+    getRowSize?: (params: any) => number;
+    keyMapper?: (row: any) => string;
```

Comment:
@ali-hosseini-deriv we can figure out params and row types by checking how they are used in the code or what is passed.🙂

Reply:
it can be anything as the datasource is completely dynamic.

Reply:
changed it to generic